core/store/src/flat_state.rs

```diff
@@ -129,8 +135,16 @@ mod imp {
         pub fn new_flat_state_for_shard(
```

Comment:
Add comments for what this function does, especially for `block_hash`, what it means when it is None